blueflood-core/src/main/java/com/rackspacecloud/blueflood/io/serializers/metrics/BasicRollupSerDes.java

```diff
@@ -54,74 +48,48 @@ public BasicRollup deserialize(ByteBuffer byteBuffer) {
         }
     }
 
-    protected BasicRollup deserializeV1Rollup(CodedInputStream in) throws IOException {
+    protected BasicRollup deserializeRollup(CodedInputStream in, byte version) throws IOException {
         final BasicRollup basicRollup = new BasicRollup();
-        final long count = in.readRawVarint64();
-        basicRollup.setCount(count);
 
-        if (count <= 0) {
-            return basicRollup;
-        }
+        deserializeBaseRollup( basicRollup, in, version );
 
-        for (int i = 0; i < BasicRollup.NUM_STATS; i++) {
-            byte statType = in.readRawByte();
-            switch (statType) {
-                case Constants.AVERAGE:
-                    averageStatDeSer.deserialize(basicRollup.getAverage(), in);
-                    break;
-                case Constants.VARIANCE:
-                    varianceStatDeSer.deserialize(basicRollup.getVariance(), in);
-                    break;
-                case Constants.MIN:
-                    minStatDeSer.deserialize(basicRollup.getMinValue(), in);
-                    break;
-                case Constants.MAX:
-                    maxStatDeSer.deserialize(basicRollup.getMaxValue(), in);
-                    break;
-                default:
-                    throw new SerializationException("invalid stat v1 type: " + (int) statType);
-            }
+        if( version == VERSION_2_ROLLUP ) {
+            basicRollup.setSum( in.readDouble() );
         }
+
         return basicRollup;
     }
 
-    protected int sizeOf(BasicRollup basicRollup) {
-        int sz = sizeOfSize();
-        sz += CodedOutputStream.computeRawVarint64Size(basicRollup.getCount());
-        if (basicRollup.getCount() > 0) {
-            sz += averageStatDeSer.sizeOf(basicRollup.getAverage());
-            sz += varianceStatDeSer.sizeOf(basicRollup.getVariance());
-            sz += minStatDeSer.sizeOf(basicRollup.getMinValue());
-            sz += maxStatDeSer.sizeOf(basicRollup.getMaxValue());
-        }
+    protected int sizeOf(BasicRollup basicRollup, byte version) {
+
+        int sz = sizeOfBaseRollup( basicRollup );
+        if( version == VERSION_2_ROLLUP )
+            sz += CodedOutputStream.computeDoubleSizeNoTag(basicRollup.getSum());
+
         return sz;
     }
 
     protected void serializeRollup(BasicRollup basicRollup, byte[] buf) throws IOException {
         rollupSize.update(buf.length);
         CodedOutputStream protobufOut = CodedOutputStream.newInstance(buf);
-        serializeRollup(basicRollup, protobufOut);
-    }
+        protobufOut.writeRawByte(Constants.VERSION_2_ROLLUP);
 
-    protected void serializeRollup(BasicRollup basicRollup, CodedOutputStream protobufOut) throws IOException {
-        protobufOut.writeRawByte(Constants.VERSION_1_ROLLUP);
-        protobufOut.writeRawVarint64(basicRollup.getCount());          // stat count
+        serializeBaseRollupHelper( basicRollup, protobufOut );
 
         if (basicRollup.getCount() > 0) {
```

Comment:
Should this be a different check? What if basicRollup.getCount() is only 4 (only has avg, variance, min, max) ?